sdk/packages/flight-sql-client/src/lib.rs

```diff
@@ -29,14 +29,14 @@ use crate::flight_client::{execute_flight, setup_client, ClientOptions};
 
 fn init_logging() {
     // Set up a subscriber that logs to stdout
-    let subscriber = tracing_subscriber::FmtSubscriber::builder()
+    if let Err(result) = tracing_subscriber::FmtSubscriber::builder()
         .with_env_filter(
             EnvFilter::try_from_default_env().unwrap_or_else(|_| EnvFilter::new("info")),
         )
-        .finish();
-
-    // Set the global default subscriber
-    tracing::subscriber::set_global_default(subscriber).expect("Failed to set tracing subscriber");
+        .try_init()
+    {
+        warn!("Tracing already initialized: {}", result);
```

Comment:
do we know that the only possible error is already initialized?  Is there a specific error code we can check to confirm and only swallow already initialized but Panic if there's a different type of failure?

Reply:
Good point. Updated. Though there are other ways, a string match appears to be the simplest way to check.